src/core/tUnification.ml

```diff
@@ -427,27 +432,28 @@ let direct_access = function
 	| AccNo | AccNever | AccNormal | AccInline | AccRequire _ | AccCtor -> true
 	| AccCall -> false
 
-let unify_kind k1 k2 =
+let unify_kind ?(strict:bool = false) k1 k2 =
```

Comment:
Can we make this non-optional? There are only so many calls to `unify_kind` in the codebase.